elide-core/src/main/java/com/yahoo/elide/security/executors/ActivePermissionExecutor.java

```diff
@@ -197,6 +200,29 @@ public ActivePermissionExecutor(final com.yahoo.elide.core.RequestScope requestS
     }
 
     /**
+     * Get permission filter on an entity.
+     *
+     * @param resourceClass Resource class
+     * @param annotationClass Annotation class
+     * @param <A> type parameter
+     */
+    public <A extends Annotation> FilterExpression getPermissionFilter(Class<?> resourceClass,
+                                                                       Class<A> annotationClass) {
+        if (requestScope.getSecurityMode() == SecurityMode.SECURITY_INACTIVE) {
+            return null; // Bypass
+        }
+        EntityDictionary dictionary = requestScope.getDictionary();
+        FilterExpressionVisitor fev = new FilterExpressionVisitor(dictionary, requestScope);
+        ParseTree classPermissions = dictionary.getPermissionsForClass(resourceClass, annotationClass);
```

Comment:
I don't think this is correct.  Look at how read permissions works with 

expressions = expressionBuilder.buildAnyFieldExpressions(resource, annotationClass, changeSpec);

We have to look at more than just the class, but rather the union of the class and field permissions to evaluate.